lib/datastore.js

```diff
@@ -15,44 +60,167 @@ class DataStore {
    * Creates a new DataStore.
    *
    * @param {Onject} cfg - The configuration parameters
+   * @param {Object} prompts - The prompt handling functions to
+   * @param {Function} cfg.prompts.unlock - The prompt function called
+   *        to handle {unlock}
    */
   constructor(cfg) {
-    let inst = instance.stage(this);
-    inst.config = Object.assign({}, cfg);
-    inst.items = new Map();
-    inst.locked = true;
+    cfg = cfg || {};
+
+    let self = instance.stage(this);
+    self.items = new Map();
+
+    // TESTING ONLY: accept an (encrypted) empty item keys map
+    self.keystore = cfg.keys;
+    self.prompts = Object.assign({}, DEFAULT_PROMPTS, cfg.prompts);
```

Comment:
It's up to you, but we could use the object spread operator here: `{...DEFAULT_PROMPTS, ...cfg.prompts}`. I've been using that in the desktop extension.